### PR TITLE
MGMT-8835: remove clients.Interface from controller package

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
Moves the code of the controlller package to use an internal
field of the Reconciler instead of the global clients.Interface
variable.
Complete removal of the clients.Interface from the code

### CODE REVIEW

File: controllers/specialresource_controller.go [hunk 1]
```diff
@@ -77,6 +77,7 @@ type SpecialResourceReconciler struct {
 	Storage     storage.Storage
 	KernelData  kernel.KernelData
 	ProxyAPI    proxy.ProxyAPI
+	KubeAPI     clients.ClientsInterface
```
Comment (qbarrand) [inline controllers/specialresource_controller.go]: How about calling that variable `kubeClient` to be consistent with what we did in some packages?

Comment (yevgeny-shnaidman) [reply controllers/specialresource_controller.go]: fixed

**Pull request merged**